### ISSUE TITLE
Actor profile has internal domains for publicKey id and owner.

### ISSUE THREAD

**Issue by steve-bate** [xwiki-contrib/application-activitypub#37]
Hello,

I'm new to XWiki so maybe I'm doing something incorrectly. However, I'm relatively familiar with ActivityPub. 

My installation is using docker-compose and is running behind a reverse proxy (Cloudflare Tunnels).

When I deference my XWiki user-specific AP Actor's endpoint, I see public domains being used for the URLs in most fields like id, inbox, outbox, and so on. However, for publicKey id and owner, the URL is using the internal (LAN) domain name.

The XWiki AP Actor data is also missing the JSON-LD context for "https://w3id.org/security/v1" which defines the terms for publicKey and related data.

I am unable to follow the XWiki actor from Mastodon, although it's not clear to me that either of these problems is the cause. I've experimented with a custom ActivityPub server to test the domain name and context issues and Mastodon can still follow my actor.

**Comment by steve-bate** [comment]
After digging further, I discovered why Mastodon was not finding my XWiki user. In my attempt to workaround the non-standard webfinger endpoint, I added the webfinger data to another server. People sometimes do this to get redirects from personal domain names to a shared AP server. Today I learned that Mastodon will do a webfinger double-check using a webfinger server at the domain of the resolved actor. Since I was using Cloudflare Tunnels without Nginx and an XWiki webfinger redirect, the second webfinger query to my XWiki container failed.

Unfortunately, after I configured an Nginx reverse proxy, I had a couple of new problems. First, it appears that the URL scheme described in the [docs](https://extensions.xwiki.org/xwiki/bin/view/Extension/ActivityPub%20Application/#HURLScheme) isn't accurate. After trying many variations, I found that `http://serverUrl/webfinger/?resource={actorId}@{domain}` does work (no `/xwiki/` in the URL). The second problem was that `{domain}` only worked with the internal LAN domain and not the public reverse-proxied domain. Using the public domain returned "No user found for the given domain".

At this point, I think I'm stuck unless someone has some suggestions (or I have time to refresh my Java skills and create a PR). I'm going to create a few separate issues to make these easier to track.
